package/MDAnalysis/coordinates/base.py

```diff
@@ -1306,22 +1307,23 @@ def check_slice_indices(self, start, stop, step):
 
         if step > 0 and stop < start:
             raise IndexError("Stop frame is lower than start frame")
+        elif step > 0 and stop >= nframes:
```

Comment:
The exception above should not be raised; instead it should iterate over no frame.

```python
range(4)[4:2:1]  # []
```